### PR TITLE
Zhangl/issue900

### PR DESCRIPTION
**Pull request by LiilyZhang**
For issue 900.
- `hzn dev service new` will also create a template `service.policy.json` under `/horizon` folder
Screen Shot 2019-06-05 at 13 59 56

- `hzn exchange service updatepolicy "bp@us.ibm.com/lily.test_0.0.1_amd64"  -f service.policy.json` will update service policy with built-in properties:
1) all values are coming from exchange service
Screen Shot 2019-06-05 at 21 12 34

Screen Shot 2019-06-05 at 21 12 58

2) When service version is set to `0.0.2` in `service.policy.json`, the policy still using value from exchange service
Screen Shot 2019-06-05 at 21 18 54
Screen Shot 2019-06-05 at 21 48 05

### CODE REVIEW

File: cli/dev/servicepolicy.go [hunk 1]
```diff
@@ -6,10 +6,17 @@ import (
 )
 
 const SERVICE_POLICY_FILE = "service.policy.json"
+const (
+	PROP_SVC_URL     = "openhorizon.service.url"     // The unique name of the service.
+	PROP_SVC_NAME    = "openhorizon.service.name"    // The unique name of the service.
+	PROP_SVC_ORG     = "openhorizon.service.org"     // The multi-tenant org where the service is defined. If service.url is specified but this property is omitted, the org defaults to the org of the node, service or policy which is referring to the service.
+	PROP_SVC_VERSION = "openhorizon.service.version" // The version of a service using the same semantic version syntax.
+	PROP_SVC_ARCH    = "openhorizon.service.arch"    // The hardware architecture of the node this service can run on.
+)
```
Comment (linggao) [inline cli/dev/servicepolicy.go]: These are defined in externalpolicy.builtin_properties.go . Please do not redefine them.

File: cli/exchange/service.go [hunk 2]
```diff
@@ -476,14 +476,38 @@ func ServiceUpdatePolicy(org string, credToUse string, service string, jsonFileP
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, "Incorrect policy format in file %s: %v", jsonFilePath, err)
 	}
 
+	// Set default built in properties before publishing to the exchange
+	properties := policyFile.Properties
+	for i, property := range properties {
+		// remove the {"name":"prop1", "value":"value1"} from the generated service.policy.json template and set defualt built in properties
+		if property.Name == "prop1" {
+			properties = append(properties[:i], properties[i+1:]...)
+		}
```
Comment (linggao) [inline cli/exchange/service.go]: Please do not remove any user defined properties.

File: cli/exchange/service.go [hunk 3]
```diff
@@ -476,14 +476,38 @@ func ServiceUpdatePolicy(org string, credToUse string, service string, jsonFileP
...
+
+	}
+
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_URL, "$SERVICE_NAME"), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_NAME, "$SERVICE_NAME"), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_ORG, "$HZN_ORG_ID"), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_VERSION, "$SERVICE_VERSION"), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_ARCH, "$ARCH"), false)
```
Comment (linggao) [inline cli/exchange/service.go]: You need to get the service and replace SERVICE_NAME, HZN_ORG_ID, SERVICE_VERSION and ARCH with real values from the service.

File: cli/exchange/service.go (hunk 2 above)
Comment (LiilyZhang) [reply cli/exchange/service.go]: 👍 I will remove this block and keep whatever inside service.policy file

File: cli/dev/servicepolicy.go (hunk 1 above)
Comment (LiilyZhang) [reply cli/dev/servicepolicy.go]: hm, those are removed from the code already

File: cli/exchange/service.go (hunk 3 above)
Comment (LiilyZhang) [reply cli/exchange/service.go]: you mean something like `properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_NAME, getEnvs("$SERVICE_NAME")), false)`?

File: cli/exchange/service.go [hunk 4]
```diff
@@ -478,19 +478,11 @@ func ServiceUpdatePolicy(org string, credToUse string, service string, jsonFileP
...
-	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_URL, "$SERVICE_NAME"), false)
-	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_NAME, "$SERVICE_NAME"), false)
-	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_ORG, "$HZN_ORG_ID"), false)
-	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_VERSION, "$SERVICE_VERSION"), false)
-	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_ARCH, "$ARCH"), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_URL, os.Getenv("SERVICE_NAME")), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_NAME, os.Getenv("SERVICE_NAME")), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_ORG, os.Getenv("HZN_ORG_ID")), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_VERSION, os.Getenv("SERVICE_VERSION")), false)
+	properties.Add_Property(externalpolicy.Property_Factory(externalpolicy.PROP_SVC_ARCH, os.Getenv("ARCH")), false)
```
Comment (linggao) [inline cli/exchange/service.go]: please first get the service using  `cliutils.ExchangeGet` (see ServcieList function in the same file),  then use the what’s in the service to fill in the built-in policies.  If the service does not exist, return an error.

**Pull request merged**